pkg/test/local/pilot.go

```diff
@@ -0,0 +1,178 @@
+package local
+
+import (
+	"istio.io/istio/pilot/pkg/bootstrap"
+	envoyv1 "istio.io/istio/pilot/pkg/proxy/envoy/v1"
+	envoyv2 "istio.io/istio/pilot/pkg/proxy/envoy/v2"
+	meshconfig "istio.io/api/mesh/v1alpha1"
+	"istio.io/istio/pilot/pkg/config/memory"
+	"istio.io/istio/pilot/pkg/model"
+	configmonitor "istio.io/istio/pilot/pkg/config/monitor"
+	"istio.io/istio/pilot/pkg/serviceregistry/aggregate"
+	"fmt"
+	"os"
+	"google.golang.org/grpc"
+	middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
+	"strconv"
+	"istio.io/istio/pilot/pkg/networking/core/v1alpha3"
+	"istio.io/istio/pilot/pkg/networking/plugin/registry"
+	"net/http"
+	"net"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	defaultGrpcMaxStreams = 100000
+)
+
+type args struct {
+	Namespace string
+	Mesh *meshconfig.MeshConfig
+	ServiceControllers *aggregate.Controller
+	ConfigDir string
+	MixerServiceAccountName []string
+	DiscoveryServiceOptions envoyv1.DiscoveryServiceOptions
+}
+
+type Pilot struct {
+	discoveryOptions envoyv1.DiscoveryServiceOptions
+	configController model.ConfigStoreCache
+	configMonitor *configmonitor.Monitor
+	discoveryService *envoyv1.DiscoveryService
+	xdsServer   *envoyv2.DiscoveryServer
+	grpcServer *grpc.Server
+	httpServer *http.Server
+}
+
```

Comment:
There is already code to start an in-memory pilot, used in the v2 local tests.

I would prefer to minimize disruption and reuse existing options.